teleprox/tests/test_rpc.py

```diff
@@ -367,19 +365,111 @@ def test_disconnect():
         cp = start_process(f'test_disconnect_client_proc{i}')
         cli = cp.client._import('teleprox').RPCClient(server_proc4.client.address)
         cp.kill()
-    
+
     start = time.time()
     server_proc4.client.close_server()
     assert time.time() - start < 1.0
     assert server_proc4.client.disconnected() == True
-    
+
     # add by Sam: force the end of process
     server_proc4.kill()
 
 
+def test_callbacks():
+    """Test proper way to pass callbacks into remote processes."""
+    logger.info("-- Test callback functionality --")
+
+    callback_result = []
+
+    def my_callback(data):
+        callback_result.append(f"callback received: {data}")
+        return "callback_response"
+
+    # First, test that callbacks fail without start_local_server
+    proc_no_server = start_process('test_callback_proc_no_server', start_local_server=False)
+
+    # Create a simple function that takes another function as parameter
+    # Use built-in functionality rather than publishing our own class
+    builtins = proc_no_server.client._import('builtins')
+
+    # This should fail because we can't serialize the function without a local server
+    try:
+        # Try to use map() with our callback - this should fail
+        result = builtins.list(builtins.map(my_callback, ["test1", "test2"]))
+        assert False, "Should have raised TypeError when sending callback without local server"
+    except TypeError:
+        pass  # Expected
+
+    proc_no_server.stop()
+
+    # Now test with start_local_server=True - this should work
+    proc_with_server = start_process('test_callback_proc_with_server')  # , start_local_server=True)
+    from teleprox.server import RPCServer
+    local_server = RPCServer()
+    local_server.run_in_thread()
+
+    # Define a simple class that can accept and invoke callbacks
+    proc_with_server.client._import("builtins").exec("""
```

Comment:
Is this used in a way that is different from the previous test? (looks like we can remove lines 411-451)